test/e2e/aws_test.go

```diff
@@ -304,18 +411,7 @@ func makeAWSMachineTemplate(namespace, name string) {
 	Expect(kindClient.Create(context.TODO(), awsMachine)).NotTo(HaveOccurred())
 }
 
-func deployCNI(tmpDir, namespace, clusterName, manifestPath string) {
-	cluster := &clusterv1.Cluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      clusterName,
-		},
-	}
-	kubeConfigData, err := kubeconfig.FromSecret(kindClient, cluster)
-	Expect(err).NotTo(HaveOccurred())
-	kubeConfigPath := path.Join(tmpDir, clusterName+".kubeconfig")
-	Expect(ioutil.WriteFile(kubeConfigPath, kubeConfigData, 0640)).To(Succeed())
-
+func deployCNI(kubeConfigPath, manifestPath string) {
```

Comment:
instead of sending `kubeConfigPath` as an arg, call the `createClusterKubeConfig` from here.

Something like:
```
func deployCNI(tmpDir, namespace, clusterName, manifestPath string) {
     createClusterKubeConfig(tmpDir, namespace, clusterName)
     .
     .

```

Reply:
I have added method to fetch kubeConfig and k8sclient which were can be used across. That will address this.